cw-orch-daemon/src/network_config.rs

```diff
@@ -0,0 +1,139 @@
+use std::collections::HashMap;
+
+use cw_orch_core::environment::{ChainInfoOwned, ChainKind, NetworkInfoOwned};
+use log::{log, Level};
+
+use crate::env::default_state_folder;
+
+pub struct NetworkConfig(HashMap<String, ChainInfoOwned>);
+
+impl NetworkConfig {
+    pub fn load() -> Option<Self> {
+        let config_path = match default_state_folder() {
+            Ok(mut state_folder) => {
+                state_folder.push("networks.json");
+                state_folder
+            }
+            Err(err) => {
+                log!(
+                    Level::Warn,
+                    "Failed to get network config {err}, using default config"
+                );
+                return None;
+            }
+        };
+        let data = match std::fs::read(config_path) {
+            Ok(data) => data,
+            Err(e) => {
+                log!(
+                    Level::Info,
+                    "Couldn't read network config file: {e}, using default config"
+                );
+                return None;
+            }
+        };
+
+        match serde_json::from_slice(&data) {
+            Ok(network_config) => Some(Self(network_config)),
+            // Only this is critical
+            Err(e) => panic!("Invalid network config: {e}"),
+        }
+    }
+
+    pub fn update_chain_info(&self, mut chain_info: ChainInfoOwned) -> ChainInfoOwned {
```

Comment:
I'm a bit confused by this function. 

When would the chain_info object be empty?

A big part of this essentially merges two chain_info objects, should we not move that implementation to the ChainInfoOwned type and rename it 

`fn overwrite_with(self, chain_info) -> Self` or similar.

Reply:
> I'm a bit confused by this function.
> 
> When would the chain_info object be empty?

`ChainInfo::config(chain_id)` creates empty chain info, except chain_id

> A big part of this essentially merges two chain_info objects, should we not move that implementation to the ChainInfoOwned type and rename it
> 
> `fn overwrite_with(self, chain_info) -> Self` or similar.

Ok!